### PR TITLE
Add types for stream-to-string

### PR DESCRIPTION
**Pull request by BendingBender**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/stream-to-string/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+export = streamToString;
+
+// tslint:disable:unified-signatures
+
+declare function streamToString(stream: NodeJS.ReadableStream): Promise<string>;
+declare function streamToString(stream: NodeJS.ReadableStream, cb: Callback): Promise<string>;
```
Comment (uniqueiniquity) [inline types/stream-to-string/index.d.ts]: If you just make this `cb?: Callback` then you can get rid of the first overload and (I think) also the tslint disable.

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

File: types/stream-to-string/index.d.ts (hunk 1 above)
Comment (BendingBender) [reply types/stream-to-string/index.d.ts]: The problem is that I don't want to mix the signature of the `enc` parameter with the `cb` parameter. I find this version much less comprehensible than the current one:
```
declare function streamToString(
    stream: NodeJS.ReadableStream,
    enc?: Callback | string,
    cb?: Callback
): Promise<string>;
```
It is just not clear what happens here and what parameters go where. I know that tecnically, this is the same.

Comment (uniqueiniquity) [reply types/stream-to-string/index.d.ts]: I wouldn't suggest combining all three into one, but just the first two:
```ts
declare function streamToString(stream: NodeJS.ReadableStream, cb?: Callback): Promise<string>;
declare function streamToString(
    stream: NodeJS.ReadableStream,
    enc: string,
    cb?: Callback
): Promise<string>;
```

This way it's clear that there's a version with one param, one with two where the second is cb, one with two where the second is enc, and one with three.

Comment (BendingBender) [reply types/stream-to-string/index.d.ts]: Oh, I didn't try this one. Fixed.

**Review by uniqueiniquity — APPROVED** [review]

**Pull request merged**